controller/sdk-set.ts

```diff
@@ -55,6 +53,76 @@ class SDKSetController {
       res.status(CODE.INTERNAL_SERVER_ERROR).send(RESPONSE.INTERNAL_SERVER_ERROR)
     }
   }
+
+  public static async createFranchiseHistory(_req: Request, res: Response, _next: NextFunction) {
+    try {
+      const franchiseCall = async () => {
+        try {
+          const res = await axios.get(SDK_DIR.FRANCHISE_HISTORY)
+          return res.data.FranchiseHistory
+        } catch (error: any) {
+          goodlog.error(error)
+          res.status(CODE.INTERNAL_SERVER_ERROR).send(RESPONSE.INTERNAL_SERVER_ERROR)
+        }
+      }
+
+      const franchiseData = await franchiseCall()
+      const teamHistoriesByTeamId: { [key: string]: any[] } = {}
+
+      for (const f of franchiseData) {
+        const franchiseLeague = await League.findOne({ apiCode: f.LEAGUE_ID })
+
+        if (!franchiseLeague) {
+          goodlog.log(MESSAGE.NOT_FOUND)
+          continue
+        }
+
+        const teamHistory = {
+          apiCode: f.TEAM_ID,
+          league: franchiseLeague._id,
+          teamCity: f.TEAM_CITY,
+          teamName: f.TEAM_NAME,
+          founded: f.START_YEAR,
+          history: f.START_YEAR + ' - ' + f.END_YEAR,
+          years: f.END_YEAR - f.START_YEAR,
+          games: f.GAMES,
+          wins: f.WINS,
+          losses: f.LOSSES,
+          playoffAppearances: f.PO_APPEARANCES,
+          divisionTitles: f.DIV_TITLES,
+          conferenceTitles: f.CONF_TITLES,
+          championships: f.LEAGUE_TITLES
+        }
+
+        if (!teamHistoriesByTeamId[f.TEAM_ID]) {
+          teamHistoriesByTeamId[f.TEAM_ID] = []
+        }
+
+        teamHistoriesByTeamId[f.TEAM_ID].push(teamHistory)
+      }
+
+      const finalTeamHistories = Object.values(teamHistoriesByTeamId).map((histories) => {
+        if (histories.length === 1) {
+          return histories[0]
+        }
+
+        histories.sort((a, b) => b.years - a.years)
+        const mainHistory = histories[0]
+        mainHistory.teamHistory = histories.slice(1)
```

Comment:
this will not be an objectId, but i think this will work better. Change the type in the interface.

Reply:
Yeah, thought about that.